### PR TITLE
expose the seek operations in KafkaConsumer

### LINKED ISSUE #1195: Allow Kafka consumer to specify the initial offset
The idea is to allow the application to define which offset should be read first.
This can be done using the `seek` method on the Kafka Consumer.

CC @Ladicek that's what we discussed.

### PR DESCRIPTION
**Pull request by Ladicek**
Resolves #1195

### CODE REVIEW

**Comment by Ladicek** [comment]
Draft because more discussion is needed.

**Comment by Ladicek** [comment]
Based on the discussion in #1195, this commit only exposes the `seek` operations through the asynchronous `KafkaConsumer` API and adds a short documentation section on `KafkaClientService`. Everything else is already there.

**Review by cescoffier — APPROVED** [review]

File: smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSource.java [hunk 1]
```diff
@@ -104,6 +105,10 @@ public KafkaSource(Vertx vertx,
         channel = this.configuration.getChannel();
 
         // fire consumer event (e.g. bind metrics)
+        // TODO the client isn't subscribed and hence doesn't have any partitions assigned yet,
```
[the remark this answers is not in the document]
Comment (Ladicek) [reply smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSource.java]: That would be pretty close to Kafka's `auto.offset.reset`, I think?

File: smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/KafkaCDIEvents.java [hunk 2]
```diff
@@ -12,13 +12,20 @@
...
+
     @Inject
     Event<Producer<?, ?>> producerEvent;
 
     public Event<Consumer<?, ?>> consumer() {
         return consumerEvent;
     }
 
+    public Event<KafkaConsumer<?, ?>> kafkaConsumer() {
+        return kafkaConsumerEvent;
```
[the remark this answers is not in the document]
Comment (Ladicek) [reply smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/KafkaCDIEvents.java]: I thought I removed this? Maybe GitHub showed a stale copy of the PR?

File: smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSource.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ladicek) [reply smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSource.java]: Plus, this change was also removed and actually isn't present in this PR :-)

**Pull request merged**